File: test-files/cpp/good-example.cpp
#include <iostream>
#include <vector>
#include <memory>


class Calculator {
private:
    double result;


public:
    Calculator() : result(0.0) {}


    double add(double a, double b) {
        result = a + b;
        return result;
    }


    double divide(double a, double b) {
        if (b == 0.0) {
            throw std::invalid_argument("Division by zero");
        }
        result = a / b;
        return result;
    }
    


    double getResult() const {
        return result;
    }
};


int main() {
    Calculator calc;
   
    std::cout << "Calculator Test" << std::endl;
    std::cout << "5 + 3 = " << calc.add(5, 3) << std::endl;
    std::cout << "15 / 3 = " << calc.divide(15, 3) << std::endl;
   
    return 0;
}

File: test-files/rust/good-example.rs
pub struct Calculator {
    result: f64,
}

impl Calculator {
    pub fn new() -> Self {
        Calculator { result: 0.0 }
    }

    pub fn add(&mut self, a: f64, b: f64) -> f64 {
        self.result = a + b;
        self.result
    }

    pub fn divide(&mut self, a: f64, b: f64) -> Result<f64, String> {
        if b == 0.0 {
            return Err(String::from("Division by zero"));
        }
        self.result = a / b;
        Ok(self.result)
    }
}

fn main() {
    let mut calc = Calculator::new();
    let sum = calc.add(5.0, 3.0);
}